Project/DAQ_System/Program/Src/Platform/I2C/STM/i2cHalWrapper_stmf4.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * i2cHalWrapper_stmf4.cpp
+ *
+ *  Created on: Mar 4, 2024
+ *      Author: Nico
+ */
+#include "i2cHalWrapper_stmf4.hpp"
+
+namespace platform {
+
+	i2cHalWrapperStmf4::i2cHalWrapperStmf4(I2C_HandleTypeDef &hi2c, uint8_t SAD)
+			: bx_i2c_(hi2c) {
+		slave_add_ = SAD;
+	}
+
+	i2cHalWrapperStmf4::~i2cHalWrapperStmf4(){}
+
+	void i2cHalWrapperStmf4::Master_Transmit(uint8_t *commands){
+		HAL_I2C_Master_Transmit(&bx_i2c_,slave_add_, commands, kMaxBytes, 0xFFFFFFFF);
```

Comment:
Instead of using 0xFFFFFFFF I probably would just consider using HAL_MAX_Delay since that would make it more clear what this parameter actually does in function. Also just for future considerations maybe using a max delay for a timeout might not be a good idea. I lowkey never had a problem with it but I could easily see it being a problem as these are blocking functions that would theoretically block for a fat minute.

Reply:
I think it would be a great idea to make the timeout value a function parameter, with a default value of `HAL_MAX_Delay`.